config/src/main/java/org/springframework/security/config/annotation/authentication/configuration/InitializeAuthenticationProviderBeanManagerConfigurer.java

```diff
@@ -94,25 +95,26 @@ private <T> List<BeanWithName<T>> getBeansWithName(Class<T> type) {
 			String[] beanNames = InitializeAuthenticationProviderBeanManagerConfigurer.this.context
 				.getBeanNamesForType(type);
 			for (String beanName : beanNames) {
-				T bean = InitializeAuthenticationProviderBeanManagerConfigurer.this.context.getBean(beanName, type);
-				beanWithNames.add(new BeanWithName<T>(bean, beanName));
+				Supplier<T> bean = () -> InitializeAuthenticationProviderBeanManagerConfigurer.this.context
```

Comment:
Instead of placing a supplier here, could we instead change the configure method above in the following two ways:

1. Don't collect the beans; instead call `ApplicationContext#getBeanNamesForType` - this should eliminate `BeanWithName` and `getBeansWithName` altogether
2. Look up the actual bean after the validation passes